config/config.go

```diff
@@ -862,28 +870,18 @@ func validateProcessors(cfg *configmodels.Config) {
 // It also expands all the string values.
 func getConfigSection(v *viper.Viper, key string) *viper.Viper {
 	// Unmarshal only the subconfig for this processor.
-	sv := v.Sub(key)
-	if sv == nil {
-		// When the config for this key is empty Sub returns nil. In order to avoid nil checks
-		// just return an empty config.
-		return viper.New()
-	}
+	sv := viperSub(v, key)
+	expandEnvConfig(sv)
 	// Before unmarshaling first expand all environment variables.
-	return expandEnvConfig(sv)
+	return sv
 }
 
 // expandEnvConfig creates a new viper config with expanded values for all the values (simple, list or map value).
 // It does not expand the keys.
-// Need to copy everything because of a bug in Viper: Set a value "map[string]interface{}" where a key has a ".",
```

Comment:
I assume there is already a test covering this case and the change is already covered by that.

Reply:
Yes.